### PR TITLE
FIX - checkSig was returning true on arbitrary confirmation signatures

### PR DESCRIPTION
**Pull request by bh2smith**
Fix for #17

### CODE REVIEW

File: contracts/Validate.sol [hunk 1]
```diff
@@ -9,7 +9,7 @@ import "openzeppelin-solidity/contracts/ECRecovery.sol";
  * @dev Checks that the signatures on a transaction are valid
  */
 library Validate {
-    function checkSigs(bytes32 txHash, bytes32 rootHash, uint256 inputCount, bytes sigs)
+    function checkSigs(bytes32 txHash, bytes32 rootHash, uint inputCount, bytes sigs)
```
Comment (fleupold) [inline contracts/Validate.sol:12]: Is this change intended? If we want to save 3 characters we should do it in the entire codebase, but I personally prefer the explicit mention of size (given that all other in types have it as well).

File: test/validate.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ contract('Validate', (accounts) => {
     const zeroHash32 = "0x0000000000000000000000000000000000000000000000000000000000000000"
 
     const short_sig = "0x7226EB9E77E58C172BDC510C76E15B4BF9A81EF"
-    const long_sig = "0x7226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F516"
+    const too_long_sig = "0x7226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5162BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A82BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A82BDC510C76E15B4BF9A81EF38D91D54173DC79D135A80F5167226EB9E77E58C172BDC510C76E15B4BF9A8"
```
Comment (fleupold) [inline test/validate.js]: Is this require to look like an actual signature, or could it also be "0"*261? If so I'd prefer that because it makes it much easier to understand what you are testing.

Also, if this variable is only used in one test, we can just create it there.

**Review by josojo — APPROVED** [review]
good

File: test/validate.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ contract('Validate', (accounts) => {
...
 
     zeroSig195 = "0x000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```
Comment (fleupold) [inline test/validate.js]: nit:
`'0x' + '0'.repeat(195)`

File: test/validate.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ contract('Validate', (accounts) => {
...
     zeroSig260 = "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
```
Comment (fleupold) [inline test/validate.js]: ditto

**Review by fleupold — APPROVED** [review]

File: test/validate.js (hunk 3 above)
Comment (bh2smith) [reply test/validate.js]: Thanks, I have been waiting for some string manipulation suggestions.

File: contracts/Validate.sol (hunk 1 above)
Comment (bh2smith) [reply contracts/Validate.sol:12]: Lets vote. This is going to mean that everything has to be changed back at some point.
